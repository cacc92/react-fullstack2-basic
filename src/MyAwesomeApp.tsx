// export function MyAwesomeApp() {
//   return (
//     <>
//       <h1>César</h1>
//       <h3>Carrasco</h3>
//     </>
//   );
// }

import type { CSSProperties } from "react";

const firstName = "César";
const lastName = "Carrasco";
const favoriteGames = ["Call of Duty", "Battlefield", "Delta Force"];
const isActive = true;
const address = {
  zipCode: "ABC-123",
  country: "Chile",
};

const myStyle: CSSProperties = {
  backgroundColor: "#fafafa",
  padding: 10,
  borderRadius: 10,
  marginTop: 10,
};
// Tambien se puede resolver con una funcion tipo flecha
export const MyAwesomeApp = () => {
  return (
    <>
      <h1>{firstName}</h1>
      <h3>{lastName}</h3>

      <p>{favoriteGames.join(", ")}</p>

      <h1>{isActive ? "Esta activo" : "No esta activo"}</h1>

      <p style={myStyle}>{JSON.stringify(address)}</p>
    </>
  );
};
