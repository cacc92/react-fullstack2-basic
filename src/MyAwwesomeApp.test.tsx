import {describe, expect, test} from 'vitest'
import { MyAwesomeApp } from './MyAwesomeApp'
import {render, screen} from '@testing-library/react'

describe('MyAwesomeApp', ()=>{
    test('Debe realizar render de nombre y apellido', () => {
        //! 1 - Arrange
        // const render = <MyAwesomeApp />
        // console.log(render)
        const {container} = render(<MyAwesomeApp />);
        // console.log(container.innerHTML);
        // screen.debug()

        //! 2- Act
        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');
        // console.log(h1?.innerHTML);

        //! 3- Asserts 
        expect(h1?.innerHTML).toContain("César");
        expect(h3?.innerHTML).toBe("Carrasco")

    });

    test("Should render firstname and lastname - screen", () =>{
        const name = "César";
        render(<MyAwesomeApp />);
        // screen.debug()

        // const h1 = screen.getByRole('heading', {level: 1});
        const h1 = screen.getByTestId("first-name-title");
        console.log(h1.innerHTML)

        expect(h1.innerHTML).toContain(name)
    
    });

    test("should match snapshot - screen", () => {
        render(<MyAwesomeApp />);

        expect(screen.getByTestId("div-app")).toMatchSnapshot()
    });
});