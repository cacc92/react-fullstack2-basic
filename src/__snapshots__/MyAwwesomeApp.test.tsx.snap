// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot - screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
     
    CÃ©sar
     
  </h1>
  <h3>
    Carrasco
  </h3>
  <p>
    Call of Duty, Battlefield, Delta Force
  </p>
  <p
    style="background-color: rgb(250, 250, 250); padding: 10px; border-radius: 10px; margin-top: 10px;"
  >
    {"zipCode":"ABC-123","country":"Chile"}
  </p>
</div>
`;
